### PR TITLE
Update documentation with a CUDA example.

### PR DESCRIPTION
**Pull request by proteneer**

### CODE REVIEW

File: content/getting_started/CUDAGuide.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
+---
+title: "CUDA Guide"
+date: "2021-01-20"
+draft: false
+weight: 5
+---
+
+## Reference C++ example
+
+Suppose we wanted to port the following C++ code to CUDA, with Enzyme autodiff support:
```
Comment (wsmoses) [inline content/getting_started/CUDAGuide.md:10]: Can you add a note here saying that support CUDA is still in progress research and thus experimental.

File: content/getting_started/CUDAGuide.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    printf("%f %f\n", host_x, host_y);
+    printf("%f %f\n", host_d_x, host_d_y);
+
+}
+```
+
+The one-liner compilation step is (against sm_70):
+
+```sh
+clang test3.cu -Xclang -load -Xclang /path/to/ClangEnzyme-11.so -O2 -fno-vectorize -fno-unroll-loops -fPIC --cuda-gpu-arch=sm_70 -lcudart -L/usr/local/cuda-10.1/lib64
```
Comment (wsmoses) [inline content/getting_started/CUDAGuide.md:130]: Can you add a warning here that loading Enzyme into clang in this manner may not properly nest Enzyme between optimization passes, which may have a performance impact. Explicitly calling it via opt (in a similar manner to the regular usage guide) allows a proper nesting, but is more annoying to set up.

**Review by wsmoses — CHANGES_REQUESTED** [review]
LGTM with a couple of nits just warning people of experimental support.

File: content/getting_started/CUDAGuide.md (hunk 2 above)
Comment (proteneer) [reply content/getting_started/CUDAGuide.md:130]: I've actually tried to this explicitly, I noticed that two intermediate ll files get produced via

```
clang test3.cu -S -emit-llvm -O2 -fno-vectorize -fno-slp-vectorize -fno-unroll-loops --cuda-gpu-arch=sm_70 -fPIC
```
```
ls
test3.ll test3-cuda-nvptx64-nvidia-cuda-sm_70.ll
```
I followed the instructions in the Getting Started guide, and built them separately:

```
opt test3.ll -load=/home/yutong/Code/Enzyme/enzyme/build/Enzyme/LLVMEnzyme-11.so -enzyme -S -o test3_output.ll
opt test3_output.ll -load=/home/yutong/Code/Enzyme/enzyme/build/Enzyme/LLVMEnzyme-11.so -enzyme -o test3_opt.ll -S

opt test3-cuda-nvptx64-nvidia-cuda-sm_70.ll -load=/home/yutong/Code/Enzyme/enzyme/build/Enzyme/LLVMEnzyme-11.so -enzyme -S -o test3-cuda_output.ll
opt test3-cuda_output.ll -load=/home/yutong/Code/Enzyme/enzyme/build/Enzyme/LLVMEnzyme-11.so -enzyme -o test3-cuda_opt.ll -S
```

However, the final object creation step results in an error that I haven't seen before:

```
13:39 $ clang -c test3_opt.ll test3-cuda_opt.ll
warning: overriding the module target triple with x86_64-unknown-linux-gnu [-Woverride-module]
'sm_70' is not a recognized processor for this target (ignoring processor)
'+ptx64' is not a recognized feature for this target (ignoring feature)
'+sm_70' is not a recognized feature for this target (ignoring feature)
'sm_70' is not a recognized processor for this target (ignoring processor)
'sm_70' is not a recognized processor for this target (ignoring processor)
'+ptx64' is not a recognized feature for this target (ignoring feature)
'+sm_70' is not a recognized feature for this target (ignoring feature)
'sm_70' is not a recognized processor for this target (ignoring processor)
fatal error: error in backend: 64-bit code requested on a subtarget that doesn't support it!
clang-11: error: clang frontend command failed with exit code 70 (use -v to see invocation)
clang version 11.0.1 (https://github.com/llvm/llvm-project.git 43ff75f2c3feef64f9d73328230d34dac8832a91)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/yutong/local/bin
clang-11: note: diagnostic msg: Error generating preprocessed source(s) - no preprocessable inputs.
```

Comment (wsmoses) [reply content/getting_started/CUDAGuide.md:130]: Yeah this process is honestly really annoying to do manually (and in fact is why the ClangEnzyme.so was created). If you compile a regular cuda code with -v you'll see a bunch of steps necessary to successfully compile / separately link the cpu and cuda codes together. I don't think it's worth suggesting anyone take this path, but just giving a sentence explaining that there might be performance implications as a result (which can be remedied by either properly injective Enzyme using opt, or fine tuning how the ClangEnzyme loader works).

Comment (proteneer) [reply content/getting_started/CUDAGuide.md:133]: Okay thanks - I'll do this in a later pass.

**Review by wsmoses — APPROVED** [review]

**Pull request merged**